lib/OpenQA/Utils.pm

```diff
@@ -50,8 +51,31 @@ our $app;
 sub testcasedir($$) {
     my $distri  = shift;
     my $version = shift;
+sub productdir {
+    my ($distri, $version, $test_variant) = @_;
 
     my $dir = "$testcasedir/$distri";
+    if ($test_variant && -e "$dir/products/$test_variant") {
+        $dir .= "/products/$test_variant";
+    }
+    elsif (-e "$dir/products/$distri") {
+        $dir .= "/products/$distri";
+    }
+    elsif ($version && -e "$dir/products/$distri/$version") {
+        $dir .= "/products/$distri/$version";
+    }
+    # @deprecated old structure
+    else {
+        $dir .= "/$distri-$version" if $version && -e "$dir/$distri-$version";
```

Comment:
I thing you don't want concatenation here. Or else you will create `"$testcasedir/$distri/$distri-$version".